dev/com.ibm.ws.ejbcontainer.v32_fat/fat/src/com/ibm/ws/ejbcontainer/v32/fat/tests/PassivationTest.java

```diff
@@ -0,0 +1,122 @@
+
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@RunWith(FATRunner.class)
+public class PassivationTest extends FATServletClient {
+    private static final String PASSIVATION_CAPABLE_REGEXP = "@PrePassivate.*PassivationCapable(|(Anno|XML)True)Bean";
+
+    @Server("com.ibm.ws.ejbcontainer.v32.fat.passivation")
+    public static LibertyServer server;
+
+    private static boolean checkLogsAfterServerStop;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        // Use ShrinkHelper to build the Ears & Wars
+
+        //#################### PassivationCapable.war
+        WebArchive PassivationCapable = ShrinkHelper.buildDefaultApp("PassivationCapable.war", "com.ibm.ws.ejbcontainer.v32.fat.passivation.");
+        ShrinkHelper.exportDropinAppToServer(server, PassivationCapable, DeployOptions.SERVER_ONLY);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        if (server != null) {
+            server.stopServer(false, "CNTR0332W");
+            try {
+                if (checkLogsAfterServerStop) {
+                    checkLogsAfterServerStop();
+                }
+            } finally {
+                server.postStopServerArchive();
+            }
+        }
+    }
+
+    private static void checkLogsAfterServerStop() throws Exception {
+        // Check again after the server is stopped to look for passivations
+        // that erroneously occurred during application stop.
+        checkUnexpectedPrePassivate();
+
+        // The rest of the EJBs should be destroyed during application stop.
+        List<String> lines = server.findStringsInLogs("@PreDestroy");
+        Assert.assertEquals("Unexpected number of lines: " + lines.toString(), 5, lines.size());
+
+        Pattern p = Pattern.compile("@PreDestroy.*PassivationCapable(AnnoFalse|XMLFalse(|LoadStrategy(Once|Transaction|ActivitySession)))Bean");
+        for (String line : lines) {
+            Assert.assertTrue("Unexpected line: " + line, p.matcher(line).find());
+        }
+    }
+
+    private static void checkUnexpectedPrePassivate() throws Exception {
+        Pattern p = Pattern.compile(PASSIVATION_CAPABLE_REGEXP);
+        for (String line : server.findStringsInLogs("@PrePassivate")) {
+            Assert.assertTrue("Unexpected line: " + line, p.matcher(line).find());
+        }
+    }
+
+    @Rule
+    public final TestName testName = new TestName();
```

Comment:
suggestion (non-blocking): the test name rule is available via the superclass `FATServletClient`